components/core/src/clp/clp/CommandLineArguments.cpp

```diff
@@ -223,6 +226,92 @@ CommandLineArguments::parse_arguments(int argc, char const* argv[]) {
+            options_ir.add_options()(
+                    "msg-ix",
+                    po::value<size_t>(&m_ir_msg_ix)
+                            ->value_name("MSG_IX")
+                            ->default_value(m_ir_msg_ix),
+                    "Target log event index that decompressed IR must include"
+            );
+            options_ir.add_options()(
+                    "target-size",
+                    po::value<size_t>(&m_ir_target_size)
+                            ->value_name("TARGET_SIZE")
+                            ->default_value(m_ir_target_size),
+                    "Target size (B) for the IR chunk before a new chunk is created"
+            );
+            options_ir.add_options()(
+                    "temp-output-dir",
+                    po::value<string>(&m_ir_temp_output_dir)
+                            ->value_name("TEMP_OUTPUT_DIR")
+                            ->default_value(m_ir_temp_output_dir),
+                    "Temporary output directory for IR chunks"
+            );
+            po::options_description all_ir_options;
+            all_ir_options.add(ir_positional_options);
+            all_ir_options.add(options_ir);
+
+            // Parse ir decompression options
+            vector<string> unrecognized_options
+                    = po::collect_unrecognized(parsed.options, po::include_positional);
+            unrecognized_options.erase(unrecognized_options.begin());
+            po::store(
+                    po::command_line_parser(unrecognized_options)
+                            .options(all_ir_options)
+                            .positional(ir_positional_options_description)
+                            .run(),
+                    parsed_command_line_options
+            );
+
+            notify(parsed_command_line_options);
+
+            // Handle --help
+            if (parsed_command_line_options.count("help")) {
+                print_ir_basic_usage();
+
+                cerr << "Examples:" << endl;
+                cerr << "  # Decompress file with id ORIG_FILE_ID to IR " << endl;
+                cerr << "  " << get_program_name() << " i archives-dir output-dir orig-file-id"
+                     << endl;
+                cerr << endl;
+
+                po::options_description visible_options;
+                visible_options.add(options_general);
+                visible_options.add(all_ir_options);
+                cerr << visible_options << endl;
+                return ParsingResult::InfoCommand;
+            }
+
+            if (m_archives_dir.empty()) {
+                throw invalid_argument("ARCHIVES_DIR cannot be empty.");
+            }
+
+            if (m_orig_file_id.empty()) {
+                throw invalid_argument("ORIG_FILE_ID cannot be empty.");
+            }
+
+            if (m_output_dir.empty()) {
+                throw invalid_argument("OUTPUT_DIR cannot be empty.");
+            }
```

Comment:
Since these are positional options, it's important to keep the validation in the same order as the positions of the arguments. So `m_output_dir` should be validated before `m_orig_file_id`.